include/tins/exceptions.h

```diff
@@ -66,6 +66,22 @@ class malformed_packet : public exception_base {
     malformed_packet() : exception_base("Malformed packet") { }
 };
 
+/**
+ * \brief Exception thrown when a DNS decompression pointer is out of bounds.
+ */
+class DNS_decompression_pointer_out_of_bounds : public exception_base {
+public:
+    DNS_decompression_pointer_out_of_bounds() : exception_base("DNS decompression pointer out of bounds") { }
```

Comment:
nit: maybe add a colon after "decompression"? e.g. "DNS decompression: pointer out of bounds". Same for the other exception.